src/pages/index.js

```diff
@@ -0,0 +1,277 @@
+
+// Edit Form Elements
+const editModal = document.querySelector("#edit-modal");
+const editForm = editModal.querySelector(".modal__form");
+const profileFormElement = document.querySelector("#editProfileForm");
+const nameInput = document.querySelector("#edit-profile-form-name");
+const descriptionInput = document.querySelector(
+  "#edit-profile-form-description"
+);
+
+// Card Elements
+const cardList = document.querySelector(".cards__list");
+
+// Preview Elements
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+
+// Modal Elements
+const closeButtons = document.querySelectorAll(".modal__close-btn");
+const modals = document.querySelectorAll(".modal");
+const cancelButton = document.querySelector(".modal__submit-btn_type_cancel");
+
+api
+  .getAppInfo()
+  .then(([cards, userInfo]) => {
+    api.userId = userInfo._id;
+    profileAvatarElement.src = userInfo.avatar;
+    profileNameElement.textContent = userInfo.name;
+    profileDescriptionElement.textContent = userInfo.about;
+
+    cards.forEach((cardData) => {
+      const cardEl = getCardElement(cardData);
+      cardList.prepend(cardEl);
+    });
+  })
+  .catch((err) => {
+    console.error("Error:", err);
+  });
+
+let selectedCard, selectedCardId;
+
+// Functions
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keyup", escClose);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keyup", escClose);
+}
+
+function escClose(evt) {
+  if (evt.key === "Escape") {
+    const activeModal = document.querySelector(".modal_opened");
+    closeModal(activeModal);
+  }
+}
+
+modals.forEach((modal) => {
+  modal.addEventListener("mousedown", (evt) => {
+    if (evt.target.classList.contains("modal")) {
+      closeModal(modal);
+    }
+  });
+});
+
+function handleProfileFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .editUserInfo({ name: nameInput.value, about: descriptionInput.value })
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileDescriptionElement.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Save");
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .addNewCard({
+      name: cardNameInput.value,
+      link: cardLinkInput.value,
+    })
+    .then((data) => {
+      const cardEl = getCardElement(data);
+      cardList.prepend(cardEl);
+      closeModal(cardModal);
+      cardForm.reset();
+      cardSubmitBtn.disabled = true;
```

Comment:
* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have a `disabled` attribute. And the `avatar` submit button should be disabled after submission as well

You should reuse the `disableButton` function from `validation.js`, to avoid code duplication